Topics/Tech_Stacks/Vue_Component_Setup_and_Tips

```diff
@@ -0,0 +1,73 @@
+# Setting up Vue 3 Components, How to Use them, and Other Tips!
+
+## Table of Contents
+
+#### [Introduction](#Introduction)
+#### [What is a Vue Component? ](#what)
+#### [Topics](#thing)
+#### [Topics again](#something)
+#### [Other Tips](#other)
+#### [Conclusion](#conclude)
+
+## Introduction
+This is a guide to the inner workings of Vue Components and their usage. It will tell you how to set them up,
+what the file format is, and also give general advice and things to watch out for. This guide assumes the reader has basic knowledge of HTML/CSS, as well as Javascript.
+
+If you have no knowledge of Vue or how to set up a Vue project, it is imperative that you follow [Vue's Quick Start](https://vuejs.org/guide/quick-start.html) instructions for doing so.
+
+## What is a Vue Component? Why should I care? <a name="what"></a>
+The Vue.js framework is a Javascript component-based programming model for the development of User Interfaces (UI), and is commonly used for Web Applications. It builds on the syntax of HTML, allowing the user to "declaratively" describe what they want the code to do. This allows for a very reactive app, one that responds efficiently to state changes to update the DOM. What this means is, less work needs to be done by the programmer for the HTML to update. The API takes care of a lot of it!
+
+Components are how it's all done. Every .vue file is a Vue Component! In these files, you can condense the 3 key aspects of a webpage, the HTML, code-behind, and CSS styling. This centralization and user-friendliness has made Vue renowned for its simplicity.
+
+**Am I quoting too much from official documentation? Should I put my own spin on it or cut out general info about the Vue.js framework?**
+
+> To learn more, please visit the [Vue website](https://vuejs.org/guide/introduction.html).
+
+## Format of Vue Components<a name="implementation"></a>
+Here is the general format of a Vue Component.
+
+
+```c
+# Example code
+<script setup>
+import { ref } from 'vue'
+const count = ref(0)
+</script>
+
+<template>
+  <button @click="count++">Count is: {{ count }}</button>
+</template>
+
+<style scoped>
+button {
+  font-weight: bold;
+}
+</style>
+```
+The top is the Javascript code-behind, the middle is the HTML, and the bottom is the CSS style. Nice and centralized. Thankfully, the order doesn't matter. However, as the script is the most complicated, we'll talk about it last.
+
+**Should I explain what the example code above means? Like, what import {ref} does? Or maybe I should give a different example, one similar to what we worked on in our CSC301 project? I worry it might be too long**
+
```

Comment:
Example code is a good idea.

It might be sensible to split it up, though. Start by explaining just the HTML template, and showing just that part of the example.

Then the style part in the style section. Then the script part in the script section, then put it all together at the end.
The reason I ask for this is because I think it'd make sense to use as little working memory as possible -- we don't want people to have to keep the script example in memory until they get to the bottom of the page.

On example code -- assume the user knows what an import statement does -- do explain vue features that apply to components like state and stuff in your actual draft. Try to avoid going into too much detail because we could always just #read the docs if we need that.